packages/libs/eda/src/lib/core/components/filter/HistogramFilter.tsx

```diff
@@ -277,12 +277,13 @@ export function HistogramFilter(props: Props) {
   const updateUIState = useCallback(
     (newUiState: Partial<UIState>) => {
       // if (uiState.binWidth === newUiState.binWidth) return;
-      analysisState.setVariableUISettings({
+      analysisState.setVariableUISettings((currentState) => ({
```

Comment:
Was this basically broken for 2 years?
Ah, yes, it was! In the regular EDA, the subsetting panel could only remember histogram settings (log scale, bin width etc) for one variable. If you made changes to a second variable, the settings of the first would be lost.

Reply:
Quite the scandal!